platform/pddf/i2c/modules/psu/pddf_psu_module.c

```diff
@@ -0,0 +1,302 @@
+	struct i2c_board_info *board_info;
+	struct i2c_client *client_ptr;
+
+
+	/*pddf_dbg(KERN_ERR "%s: %s", __FUNCTION__, buf);*/
+
+
+	/*pddf_dbg(KERN_ERR "Creating an I2C client with parent_bus:0x%x, dev_type:%s, dev_addr:0x%x\n", cdata->parent_bus, cdata->dev_type, cdata->dev_addr);*/
+	if (strncmp(buf, "add", strlen(buf)-1)==0)
+	{
+		adapter = i2c_get_adapter(cdata->parent_bus);
+		board_info = i2c_get_psu_board_info(pdata, cdata);
+
+		/* Populate the platform data for psu */
+		/*pddf_dbg(KERN_ERR "Creating a client %s on 0x%x, platform_data 0x%x\n", board_info->type, board_info->addr, board_info->platform_data);*/
+		client_ptr = i2c_new_device(adapter, board_info);
+		
+		if(client_ptr != NULL)
+		{
+			i2c_put_adapter(adapter);
+			pddf_dbg(PSU, KERN_ERR "Created a %s client: 0x%x\n", cdata->i2c_name , client_ptr);
+			add_device_table(cdata->i2c_name, (void*)client_ptr);
+		}
+		else
+		{
+			i2c_put_adapter(adapter);
+			goto free_data;
+		}
+	}
+	else if (strncmp(buf, "delete", strlen(buf)-1)==0)
+	{
+		/*Get the i2c_client handle for the created client*/
+		client_ptr = (struct i2c_client *)get_device_table(cdata->i2c_name);
+		if (client_ptr)
+		{
+			pddf_dbg(PSU, KERN_ERR "Removing %s client: 0x%x\n", cdata->i2c_name, client_ptr);
+			i2c_unregister_device(client_ptr);
+			delete_device_table(cdata->i2c_name);
+		}
+		else
+		{
+			printk(KERN_ERR "Unable to get the client handle for %s\n", cdata->i2c_name);
+		}
+
+	}
+	else
+	{
+		printk(KERN_ERR "PDDF_ERROR: %s: Invalid value for dev_ops %s", __FUNCTION__, buf);
+	}
+
+	goto clear_data;
+
+free_data:
+	if (board_info->platform_data)
+	{
+		PSU_PDATA *psu_platform_data = board_info->platform_data;
+		if (psu_platform_data->psu_attrs)
+		{
+			printk(KERN_ERR "%s: Unable to create i2c client. Freeing the platform subdata\n", __FUNCTION__);
+			kfree(psu_platform_data->psu_attrs);
+		}
+		printk(KERN_ERR "%s: Unable to create i2c client. Freeing the platform data\n", __FUNCTION__);
+		kfree(psu_platform_data);
+	}
+
+clear_data:
+	memset(pdata, 0, sizeof(PSU_DATA));
+	/*TODO: free the data cdata->data if data is dynal=mically allocated*/
+	memset(cdata, 0, sizeof(NEW_DEV_ATTR));
+	return count;
+}
+
+
+static struct kobject *psu_kobj;
+static struct kobject *i2c_kobj;
+
+int __init pddf_data_init(void)
+{
+	struct kobject *device_kobj;
+	int ret = 0;
+
+
+	pddf_dbg(PSU, "PDDF_DATA MODULE.. init\n");
+
+	device_kobj = get_device_i2c_kobj();
+	if(!device_kobj) 
+		return -ENOMEM;
+
+	psu_kobj = kobject_create_and_add("psu", device_kobj);
+	if(!psu_kobj) 
+		return -ENOMEM;
+	i2c_kobj = kobject_create_and_add("i2c", psu_kobj);
+	if(!i2c_kobj) 
+		return -ENOMEM;
+	
+	ret = sysfs_create_group(i2c_kobj, &pddf_clients_data_group);
+	if (ret)
+	{
+		kobject_put(i2c_kobj);
+		kobject_put(psu_kobj);
+		return ret;
+	}
+	pddf_dbg(PSU, "CREATED PSU I2C CLIENTS CREATION SYSFS GROUP\n");
+
+	ret = sysfs_create_group(i2c_kobj, &pddf_psu_client_data_group);
+	if (ret)
+    {
```

Comment:
the spacing is not consistent.

Reply:
Done

Reply:
This is already taken care in commit: a654dc7.